### ISSUE TITLE
CalDAV / CardDAV problem with macOS 10.12

### ISSUE THREAD

**Issue by hejNord** [tine20/tine20#6977]
I setup a fresh tine20 installation (12.14) and found an issue with CalDAV and CardDAV with macOS Calendar:

I grab the Calendar path using the Tine20 web interface:
_https://IP/tine20/calendars/xxx/yyy_

Putting this into the browser works fine as well as ActiveSync 

The CaldDAV account was setup as follows: 
server: _IP_
serverpath _/tine20/calendars/xxx/_

macOS Calendar reports **Der Server hat einen ungültigen Kalenderort angegeben. Für den Account abc wurde kein Kalenderort angegeben.**
Forcing the Calendar to update several time, I can see the three personal calendars of user abc as well as the entries. Nevertheless the error remains and additional entries will not be synchronized. Also two little triangles beneath the calendar symbol left as well as in the headline of the calendar app remains.

My old tine20 setup (12.13) works fine using serverpath _/principals/users/xxx/_, which does not work for the current setup. I can't see any errors in the tine20 log 
Any recommendations?
Btw: Is it possible to connect tine20 as an Exchange account for macOS?

**Thx and happy new year!**

**Comment by hejNord** [comment]
I forgot to mention: The new setup (12.14) is an Debian setup which was done using apt-get install tine20 while the old one (12.13) s based on the tine20 zip-file.

**Comment by koelschejung74** [comment]
es kann gut sein, dass ein alter bug ist? Ich habe zur Zeit wieder die 2018085 Version am Laufen, der Sync (Mails, Termine und Kontakte) via Handy funktioniert. Der Sync von Terminen mit CalDav auch, aber mit den Kontakten nicht, bei gibt es Probleme mit Kontakten, die ein Bild haben.

**Comment by pschuele** [comment]
i think, osx/ical detects the possible calendars. so try just adding https://IP/tine20/calendars/ as caldav url.

> Btw: Is it possible to connect tine20 as an Exchange account for macOS?

nope, that's not possible at the moment.

> Thx and happy new year!

thanks and a happy new year to you, too :)

**Comment by hejNord** [comment]
_i think, osx/ical detects the possible calendars. so try just adding https://IP/tine20/calendars/ as caldav url._

Unfortunately this does not work but results in the same error:
<img width="609" alt="err" src="https://user-images.githubusercontent.com/46198686/50588548-ec081200-0e82-11e9-8bfb-00376f508e5c.png">

**Comment by pschuele** [comment]
hm, are your webserver rewrite rules correct? could you check if https://ip/tine20/calendars/ works in the browser (should ask for credentials and open webdav view)?

**Comment by hejNord** [comment]
Yes,  _https://ip/tine20/calendars/_ works as expected. The rewrite rules are the original rules provided by apt-get install tine20.

**Comment by pschuele** [comment]
I'll ask @corneliusweiss about this - I currently have no clue what's going wrong here.

**Comment by hejNord** [comment]
**Update**: I tried this configuration:

server: IP
serverpath /tine20/principals/users/abc123/

where abc123 is the alphanumerical string taken from the browser after entering _https://ip/tine20/principals/users/_ and selecting the correct user "user1" 

This works for me with Apple Calendar.

**Comment by hejNord** [comment]
Running the latest Tine20 version (12.18) the problem seems to be back. Entering _https://ip/tine20/principals/users/xxx_ in the browser works but CardDAV does not return any entry visible in Tine20 web interface. Anybody recommendations? Thx

**Comment by hejNord** [comment]
btw: Calendar still works for both, ActiveSync and CalDAV

**Comment by hejNord** [comment]
Update: I checked CardDAV connection to my Tine20 setup with latest Thunderbird + Cardbook Add-on running on a virtual Win7: **CardDAV works** 
Problem seems to be related to either MacOS 10.14.3 and / or the Contact App. 
Anybody else how can confirm this? Would be great if we could get this work again :)

**Comment by hejNord** [comment]
Addition test: setup CardDAV server on Synology NAS. Added account to Contacts as shown in https://idomix.de/kontakte-server-carddav-und-gemeinsam-nutzen-auf-synology-diskstation. Works fine...
So it seems to be no general issue of macOS ...

![](https://idomix.de/wp-content/uploads/2017/06/carddav.png)

**Comment by hejNord** [comment]
**Update**: Current scenario: tine20 with **two users A and B**. Create first CardDAV account in Contacts App  for user A - works. Create Second CardDAV account in Contacts App  for user B - No error but it did not work (empty, no synchronization) - any ideas?
_Latest Tine20 and latest macOS_

**Comment by pschuele** [comment]
hm, could you please attach some logs? tine20.log in DEBUG mode should give some hints about the problem.

**Comment by hejNord** [comment]
Please find attached a part of the log file. 
[log_part.log.zip](https://github.com/tine20/tine20/files/2932408/log_part.log.zip)

**Comment by hejNord** [comment]
Hope this helps

**Comment by hejNord** [comment]
Further update: I setup a second macOS system within my network. Both systems are on the same OS level. While both are able to use the Tine20 calendar via Apples Calendar App, only on is able to get the information via Apples Contacts App. Both systems can reach Tine20 via browser  _https://ip/tine20/addressbooks/_

**Comment by msphn** [comment]
@hejNord I could verify this one, CardDAV also doens't work on my setup while CalDAV is working just fine.

**Comment by msphn** [comment]
@hejNord Nope, It actually works. It took me some time and then macOS asked me for a password again and then the contacts appeared. Something is buggy, but not sure if Tine 2.0 is the problem here.

Probably something changed with the handshake but it somehow works now.

**Comment by hejNord** [comment]
Thanks for trying. I setup a new user but the effect remains the same. No additional password request appears. Are you going to keep an eye on this issue?

**Comment by msphn** [comment]
I'm not going to dig into it. Some colleagues using it with Mojave and it worked for them out of the box. Probably @pschuele can do something if you give him sufficient information on how to exactly reproduce this problem.

**Comment by corneliusweiss** [comment]
afaik apple expects the stuff to be in the root folder. Try to deliver directly from https://fqdn/ without subpath
